src/scripting/lua_map.cc

```diff
@@ -2442,13 +2490,14 @@ int LuaImmovableDescription::has_attribute(lua_State* L) {
 }
 
 /* RST
-   .. method:: probability_to_grow
+   .. method:: probability_to_grow(terrain_description)
 
-      :arg terrain: The terrain that we are checking the probability for.
-      :type terrain: :class:`wl.map.TerrainDescription`
+      Returns a :class:`double` describing the probability that this immovable will grow on the
```

Comment:
It would be good to have a note that working with `double` values is desync-prone because floating point arithmetic is hardware- and compiler-dependent. Not sure if you want to add this here.

Reply:
Currently this is only used (and was introduced for the help) using this in normal scripting might be dangerous. A Note should clarify this imho.

Reply:
Lua does not know any of the types `integer`, `uint`, `double`, `float`, … . Any numerical value in Lua is of type `number`.

So i don't see a reason why this should be desync-prone or dangerous when using this **value** in scripts. At least it would be not a  fault of a scripts author.

I think the original author of this text just wanted to say: "This value is a floating point number in a range of +x.y until -x.y", not "This is value is of type double".

What about:
> Returns a number in form of a `double` describing …

or:
> Returns a value in form of a `double` describing …

or:
> Returns a floating point number describing …

We should always think about that the Lua documentation is mainly for people who wants to do scripting. The documentation shouldn't be wrong, but also easy to understand for people who are inexperienced in programming. Difficult task though.

Reply:
The problem about doubles or floating point arithmetics is that if you do them on different platforms the results might differ, due to rounding effects resulting from the internal precision / implementation.      
If you use fractional numbers in a script the result might differ from one computer executing the script to the other which will lead to a desync. 
e.g. 300 * 0.07 < 21 might be true or false depending on the machines floating implementation, while 300 * 0.07 == 21 most probably will be false on both of them.
for this reason we got rid of floating point calculation as far as we could by simply using large integers and doing integer arithmetics on them.
in Lua every number is internally represented as double normally (therefore I assume this is the case in our implementation of lua as well). But the programmer needs to take care to ensure the results are integer. therefore especially in skripting it might be recomendable to use the integer floor division "//" operator which returns the smallest integer close to the result.

see chapter 3.4.1 of https://www.lua.org/manual/5.3/manual.html for example.
see also https://github.com/widelands/widelands/blob/3f3c18dea19ff2cfa449603398c4953ed8b59ecf/src/logic/map_objects/terrain_affinity.cc#L34

I hope that helps. 
therfore my recommendation is twofold we might test to switch this function to return integers (which will be just percentages without any fractional part e.g. 95% instead of 95.45 or 95.95%) as the precision is not needed for the usage in the terrain and immovable help imho, or we just might add a statement to warn the scripter about the possible implications of double arithmetics.

Reply:
No need to feel sorry, hessenfarmer. I apologize for my harsh words.

We all are trying to do our best for this game, each in his/her scope of experience.